colossalai/fx/profiler/profiler.py

```diff
@@ -257,18 +331,25 @@ def f(*args: Tuple[Argument, ...], **kwargs: Dict[str, Any]) -> Any:
         # If there is an argument that this `call_module` is inplace, we should
         # still run the profiling but discard some results regarding `module`.
         inplace = getattr(module, 'inplace', False)
+
+        # calculate parameter size
+        param_size = parameter_size(module)
+
         if inplace:
             module.inplace = False
         if device == 'meta':
             out, meta = _profile_meta(func, *args, **kwargs)
-        else:
-            out, meta = _profile_concrete(func, *args, **kwargs)
-        if inplace:
-            # super-dainiu: experiments on mobilenet_v2 shows that `torch.nn.ReLU`
-            # is the only inplace activation function that discard its input.
-            if type(module) in [torch.nn.ReLU]:
+
+            # currently we set the fwd_mem_tmp of ReLU to zero
+            if type(module) in [torch.nn.modules.activation.ReLU]:
                 meta.save_fwd_in = False
                 meta.bwd_mem_out = 0
+                meta.fwd_mem_tmp = 0
+        else:
+            out, meta = _profile_concrete(func, *args, **kwargs)
+
+        # grad for param will not be counted
+        meta.bwd_mem_tmp -= param_size
```

Comment:
a hypothesis here: why bwd_temp captures the grad of weights in meta profiling?

Reply:
fixed~